pumpkin/src/client/authentication.rs

```diff
@@ -143,6 +143,8 @@ pub enum AuthError {
     FailedParse,
     #[error("Unknown Status Code")]
     UnknownStatusCode(StatusCode),
+    #[error("You are already connected to this server")]
+    DuplicateConnection,
```

Comment:
I don't like to have the error here, This has not really something to do with Authentication. Is more just validation

Reply:
Do you have a spot in mind you would prefer the error be stored? In this case I also believe the duplicate UUID check should be moved out of `authenticate` and potentially go inside at the start of the `if BASIC_CONFIG.online_mode` block, line 153 in `client_packets.rs`?

Reply:
The second option makes more sense to me

Reply:
Thanks, should be resolved in [7b7eab5](https://github.com/Snowiiii/Pumpkin/pull/178/commits/7b7eab5dc67c47752c52a7b2b6aa1dcf60abf748) 🙂

Reply:
marking as resolved